File: sherlock.js
//https://www.hackerrank.com/challenges/sherlock-and-valid-string
function isValid(s) {
  let cnts = {};
  let difs = [];
  Array.from(s).forEach((c) => {
    cnts[c] = cnts[c] || 0;
    cnts[c]++;
  });
  let pivot;
  Object.keys(cnts).forEach((k) => {
    if (!pivot) pivot = cnts[k];
    if (pivot != cnts[k]) difs.push(cnts[k]);
  });
  return difs.length > 1 ? "NO" : "YES";
}

console.log(isValid("abdddde"));

File: sherlock.py
# https: // www.hackerrank.com/challenges/sherlock-and-valid-string
from collections import Counter


def isValid(s):
    freq = [v for k, v in Counter(s).items()]
    pivot = freq[0]
    different_freqs = []
    for i in freq[1::]:
        if(pivot != i):
            different_freqs.append(i)
    if(len(different_freqs) > 1):
        return 'NO'
    else:
        return 'YES'


print(isValid("aabbccdd"))
